src/hooks/usePeople.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import { useFilters } from './useFilters';
+import { Person } from '../types';
+import { getFilteredPeople } from '../utils/getFilteredPeople';
+import { getPeople } from '../api';
+import { getPeopleWithParents } from '../utils/getPeopleWithParents';
+
+export const usePeople = () => {
+  const { sex, query, centuries, sort, order } = useFilters();
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const sortSettings = { sort, order };
+  const filter = { sex, query, centuries };
+
+  const filteredPeople = getFilteredPeople(people, filter, sortSettings);
+
+  useEffect(() => {
+    setLoading(true);
+    setError(false);
+
+    getPeople()
+      .then(getPeopleWithParents)
+      .then(setPeople)
+      .catch(() => {
+        setError(true);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.